PET and VET creation

RoriCremer opened the pull request:
Exome VET 2.0  
Keeping:
    start_position, // req
    reference_bases, // req
    alternate_bases, // req
    alternate_bases.alt, // req
    alternate_bases.AS_RAW_MQ, // req
    alternate_bases.AS_MQ_DP, // req
    alternate_bases.AS_RAW_MQRankSum,
    alternate_bases.AS_QUALapprox, // req
    alternate_bases.AS_RAW_ReadPosRankSum,
    alternate_bases.AS_SB_TABLE, // req
    alternate_bases.AS_VarDP, // req
    call, // req
    call_name, // req
    call_genotype, // req
    call_AD,
    call_DP,
    call_GQ, // req
    call_PGT,
    call_PID,
    call_PL

Commit:
initial vet creation class

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+        START_POSITION("start_position") { // Required
+             public String getColumnValue(final VariantContext variant) throws IOException {
+                 final String startPosition = String.valueOf(variant.getStart());
+                 if (startPosition.equals(null)) {
```
this won't ever be true.  `variant.getStart()` returns an `int` like you already said which cant be null (it defaults to 0 if not initialized). therefor the string cant be null, I would say you dont need the check at all here because variant context will fall over if there is no start in the vcf row

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+
+        REFERENCE_BASES("reference_bases") { // Required
+            public String getColumnValue(final VariantContext variant) throws IOException {
+                final String referenceBase = variant.getReference().getBaseString();
```
you dont need to check this either because variant context will fall over.  I should have shown you the vcf spec which would have been helpful to know which required fields you should need to check and which you shouldnt. https://samtools.github.io/hts-specs/VCFv4.2.pdf  if you look at that `1.4.1 Fixed fields` section you'll see which columns are required to have values and which don't.  If they're required by teh spec then you dont need to check them here.

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+
+        ALTERNATE_BASES_ALT("alternate_bases.alt") {
+            public String getColumnValue(final VariantContext variant) throws IOException {
+                return variant.getAlternateAllele(0).getBaseString();
```
anything under the `alternate_bases` record type will return an array of values because there will be a value for each alternate base.  Depending on if we choose to output json/csv/avro we'll have to format that array of values differently.  So say we were using csv and we were respresenting an array of values as a `:` separated string.  This would want to look something like.

```
String output = ""
for(Allele a : variant.getAlternateAlleles()) {
  output += a.getBaseString() + ":"
}
return output
```

youd want to actually make this a function and make it better but you get the gist

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+
+        CALL_GENOTYPE("call.genotype") {
+            public String getColumnValue(final VariantContext variant) throws IOException {
+                return variant.getGenotype(0).getGenotypeString(); // TODO double check
```
because we flattened call in the schema this is perfect

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+        private String value;
+
+        HeaderFieldEnum(String value) {
+            this.value = value;
```
we should think about what we want the value for each enum to be.  Maybe it should be the column name in the schema?

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+        }
+
+        @Override
+        @JsonValue
```
what does this do?

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+            return String.valueOf(value);
+        }
+
+        @JsonCreator
```
same here, whats the purpose of this?

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+
+
+    public List<String> createTSV(final VariantContext variant) throws IOException {
+        List<String> rows = new ArrayList<>();
```
`row`

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+
+        for ( final HeaderFieldEnum fieldEnum : HeaderFieldEnum.values() ) {
+            final String newRow = fieldEnum.getColumnValue(variant);
+            rows.add(newRow);
```
can just be `row.add(fieldEnum.getColumnValue(variant);`

jsotobroad reviewed (commented):
BRUH

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+
+        FILTER("filter") {
+            public Set<String> getColumnValue(final VariantContext variant) throws IOException {
+                return variant.getFilters();
```
if the variant is filtered, are we just ignoring it?

jsotobroad commented on src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java:
```diff
@@ -0,0 +1,249 @@
+        },
+
+        /*ALTERNATE_BASES_AS_RAW_MQ("alternate_bases.AS_RAW_MQ") { // Required
+            if (variant.getAttribute("AS_RAW_MQ").equals(null)) {
```
ask Laura what these strings look like or look in a vcf

Commit:
initial variant walker for jg

Commit:
styling fix in unrelated code

Commit:
initial pet creation class

Commit:
add pet creation to walker

Commit:
BRUH - cleanup the cword code

Commit:
update vet schema

Commit:
add headers enum and method to pet creation

Commit:
add xsv creation to loop

Commit:
update vet

Commit:
fix extra delim bug and show non-ref placeholder

Commit:
dont rebase me bruh
Also fixed `[.....]` strings in the vet and converted a lot of strings to constants plus a couple of TODOs

Commit:
Fix GT String to output index of alleles instead of allele charaters

Commit:
fix enum bug

Commit:
add GQ band to drop as an optional param

Commit:
include end base

Commit:
code used to push PET and VET to BQ

Commit:
i dont know, it finally fuckign works....seriously.  also schema jsons

Commit:
update schema and pray

Commit:
update schema cuz im bad

Commit:
added ability to pass in what GQ state to interpret as "missing" as well as skeleton for creating the metdata tsv (sample list, what interval list was use, the interval list's md5, and what GQ is signified by "missing".  Also put some util files in a folder under BlahVariantWalkerUtils for now (should get rid of them eventually but nice to have for now)

Commit:
spacing + adding optional to GQ arguments (whoops)

Commit:
testing wip

Commit:
cleanup

Commit:
pull in argument changes

Commit:
rename intervalListPath to  intervalListBlob

Commit:
pass thru sample name

Commit:
swap out errors thrown

Commit:
add stringified  interval list to metadata table

Commit:
fixup on interval blog rename

Commit:
use L for interval list

This doesn't have the level of validation that we need, but works for now
Ideally the interval list is required to be a picard interval list file

Commit:
wip testing

Commit:
to be dropped? interval wip

Commit:
add testing materials

Commit:
wip which type should this be passed in asg

Commit:
md5 interval list concated string

Commit:
wip split xsvs by contig

Commit:
wip for testing

Commit:
wip testing

Commit:
add vet stripping intial changes

Commit:
wip to drop notes

Commit:
fix overzealous missing bug

Commit:
review vet trim with Laura

Commit:
laura seal of approval on schema

Commit:
laura seal of approval on schema

Commit:
fix substring bug

RoriCremer committed:
add back call_DP until we are sure evoq doesn't need it

RoriCremer committed:
remove unnec string

RoriCremer committed:
write chr dirs on the fly

RoriCremer committed:
create dirs for chrs w pet and vet dirs inside

RoriCremer committed:
add sample id to metadata

RoriCremer committed:
semantics for sample id

RoriCremer committed:
move position and sample into main method for easier access

RoriCremer committed:
add sample id mapping and use it

RoriCremer committed:
add enum for chr index

RoriCremer committed:
use longs for the chr addition
